pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -212,6 +212,26 @@ func (r *ReconcileDeadmansSnitchIntegration) Reconcile(request reconcile.Request
 			return reconcile.Result{}, err
 		}
 
+		if clusterdeployment.Spec.PowerState == hivev1.HibernatingClusterPowerState {
```

Comment:
This will run every reconcile loop.  At a minimum we get 2x lookup of SS, 2x lookup of secret, and a log statement every time it runs.

Recommend pull the logic in deleteSnitch for checking if a delete needs to be done out and before this block.  It could be used to control if create or delete functions are called, optimize everything a bit.